class-classic-editor-endpoint.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+class Classic_Editor_Endpoint extends WP_REST_Controller {
+	public function register_routes() {
+		$version = '1';
+		$namespace = 'classic-editor/v' . $version;
+
+		register_rest_route( $namespace, '/settings', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_user_settings' ),
+				'permission_callback' => array( $this, 'get_user_settings_permissions_check' ),
+			) ) );
+  	}
+ 
+	/**
+	* Check if a given request has access to the user's Classic Editor plugin settings.
+	* 
+	* @param WP_REST_Request $request Full data about the request.
+	* @return true|WP_Error
+	*/
+	public function get_user_settings_permissions_check( $request ) {
+		return get_current_user_id() > 0;
```

Comment:
Wondering if this would also need `current_user_can( 'edit_posts' )` as the setting is pretty pointless for "subscriber" and lower level users. 
(On the user profile screen the plugin reuses the PHP global `$user_can_edit` before showing the settings, see: https://github.com/WordPress/wordpress-develop/blob/master/src/wp-admin/user-edit.php#L73 and https://github.com/WordPress/classic-editor/blob/master/classic-editor.php#L427)

Reply:
Good point. The `$user_can_edit` global didn't seem to work for me in this context but `current_user_can( 'edit_posts' )` did.